frontend/src/components/AdminPanel/AnonymizeUsers.tsx

```diff
@@ -0,0 +1,179 @@
+import useApiGet from './useApiGet'
+import commonStyles from './common.module.css'
+import { useAppSelector } from '../../redux/hooks'
+import { selectUserState } from '../../redux/User'
+import PictureAndNameCell from './PictureAndNameCell'
+import { getAttribute } from './helpers'
+import AnonymizeUserDialog from './AnonymizeUserDialog'
+import { anonymizeUser as anonymizeUserApiCall } from './adminApi'
+import { ORGANIZATION_ID_ATTRIBUTE } from '../../constants'
+
+type UserProps = {
+  user: any
+  anonymizeUser: (user: any) => void
+}
+
+const User = ({ user, anonymizeUser }: UserProps) => {
+  const { t } = useTranslation()
+
+  const username = user.Username
+  const name = getAttribute(user, 'name')
+  const email = getAttribute(user, 'email')
+  const picture = getAttribute(user, 'picture')
+
+  return (
+    <TableRow>
+      <TableCell>
+        <PictureAndNameCell name={name} picture={picture} />
+      </TableCell>
+      <TableCell>{email}</TableCell>
+      <TableCell>{username}</TableCell>
+      <TableCell align="center">
+        <Button
+          variant="contained"
+          color="primary"
+          onClick={() => anonymizeUser(user)}
+          style={{ margin: '0 auto' }}
+        >
+          {t('admin.anonymizeUsers.anonymize')}
+        </Button>
+      </TableCell>
+    </TableRow>
+  )
+}
+
+type AnonymizeUsersTableProps = {
+  users: any[]
+  anonymizeUser: (user: any) => void
+}
+
+const AnonymizeUsersTable = ({
+  users,
+  anonymizeUser,
+}: AnonymizeUsersTableProps) => {
+  const { t } = useTranslation()
+  const userState = useAppSelector(selectUserState)
+
+  return (
+    <TableContainer className={commonStyles.tableContainer}>
+      <Table stickyHeader>
+        <TableHead>
+          <TableRow>
+            <TableCell>{t('employee')}</TableCell>
+            <TableCell>{t('email')}</TableCell>
+            <TableCell>{t('username')}</TableCell>
+            <TableCell />
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {users.map(
+            (user: any) =>
+              user.Username != userState.userName && (
+                <User
+                  key={user.Username}
+                  user={user}
+                  anonymizeUser={anonymizeUser}
+                />
+              )
+          )}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  )
+}
+
+const AnonymizeUsers = () => {
+  const { t } = useTranslation()
+  const userState = useAppSelector(selectUserState)
+
+  const {
+    result: users,
+    error: apiError,
+    loading,
+    refresh,
+  } = useApiGet({
+    getFn: listAllUsersInOrganization,
+    params: userState.organizationID,
+  })
+
+  const [userToAnonymize, setUserToAnonymize] = useState<any>()
+  const [showAnonymizeUserDialog, setShowAnonymizeUserDialog] =
+    useState<boolean>(false)
+  const [anonymizationError, setAnonymizationError] = useState<string | null>(
+    null
+  )
+
+  const anonymizeUser = (user: any) => {
```

Comment:
Jeg er litt usikker på om jeg synes dette navnet kanskje var litt forvirrende.

Jeg kommer ikke på noe bedre selv, men det høres litt ut som om denne funksjonen gjør anonymizeUser uansett (kanskje? Det er også mulig at det ikke er noen særlig bedre måter å formulere den på)

Muligens burde den kanskje bare kalles "showAnonymizeUserDialog", men jeg er åpen for diskusjon